Mathlib/Analysis/Convex/Cone/Pointed.lean

```diff
@@ -0,0 +1,200 @@
+    (S.toPointedCone hS : ConvexCone 𝕜 E) = S :=
+  rfl
+
+instance canLift : CanLift (ConvexCone 𝕜 E) (PointedCone 𝕜 E) (↑) fun S => S.Pointed where
+  prf S hS := ⟨S.toPointedCone hS, S.coe_toPointedCone hS⟩
+
+end Definitions
+
+section Maps
+
+variable [OrderedSemiring 𝕜]
+variable [AddCommMonoid E] [Module 𝕜 E]
+variable [AddCommMonoid F] [Module 𝕜 F]
+variable [AddCommMonoid G] [Module 𝕜 G]
+
+/-!
+
+## Maps between pointed cones
+
+There is already a definition of maps between submodules, `Submodule.map`. In our case, these maps
+are induced from linear maps between the ambient modules that are linear over nonnegative scalars.
+Such maps are unlikely to be of any use in practice. So, we construct some API to define maps
+between pointed cones induced from linear maps between the ambient modules that are linear over
+*all* scalars.
+
+-/
+
+/-- The image of a pointed cone under a `𝕜`-linear map is a pointed cone. -/
+def map (f : E →ₗ[𝕜] F) (S : PointedCone 𝕜 E) : PointedCone 𝕜 F :=
+  Submodule.map (f : E →ₗ[𝕜≥0] F) S
+
+@[simp, norm_cast]
+theorem toConvexCone_map (S : PointedCone 𝕜 E) (f : E →ₗ[𝕜] F) :
+    (S.map f : ConvexCone 𝕜 F) = (S : ConvexCone 𝕜 E).map f :=
+  rfl
+
+@[simp, norm_cast]
+theorem coe_map (S : PointedCone 𝕜 E) (f : E →ₗ[𝕜] F) : (S.map f : Set F) = f '' S :=
+  rfl
+
+@[simp]
+theorem mem_map {f : E →ₗ[𝕜] F} {S : PointedCone 𝕜 E} {y : F} : y ∈ S.map f ↔ ∃ x ∈ S, f x = y :=
+  Iff.rfl
+
+theorem map_map (g : F →ₗ[𝕜] G) (f : E →ₗ[𝕜] F) (S : PointedCone 𝕜 E) :
+    (S.map f).map g = S.map (g.comp f) :=
+  SetLike.coe_injective <| Set.image_image g f S
+
+@[simp]
+theorem map_id (S : PointedCone 𝕜 E) : S.map LinearMap.id = S :=
+  SetLike.coe_injective <| Set.image_id _
+
+/-- The preimage of a convex cone under a `𝕜`-linear map is a convex cone. -/
+def comap (f : E →ₗ[𝕜] F) (S : PointedCone 𝕜 F) : PointedCone 𝕜 E :=
+  Submodule.comap (f : E →ₗ[𝕜≥0] F) S
+
+@[simp, norm_cast]
+theorem coe_comap (f : E →ₗ[𝕜] F) (S : PointedCone 𝕜 F) : (S.comap f : Set E) = f ⁻¹' S :=
+  rfl
+
+@[simp]
+theorem comap_id (S : PointedCone 𝕜 E) : S.comap LinearMap.id = S :=
+  rfl
+
+theorem comap_comap (g : F →ₗ[𝕜] G) (f : E →ₗ[𝕜] F) (S : PointedCone 𝕜 G) :
+    (S.comap g).comap f = S.comap (g.comp f) :=
+  rfl
+
+@[simp]
+theorem mem_comap {f : E →ₗ[𝕜] F} {S : PointedCone 𝕜 F} {x : E} : x ∈ S.comap f ↔ f x ∈ S :=
+  Iff.rfl
+
+end Maps
+
+section PositiveCone
+
+variable (𝕜 E)
+variable [OrderedSemiring 𝕜]
+variable [OrderedAddCommGroup E] [Module 𝕜 E] [OrderedSMul 𝕜 E]
+
+/-- The positive cone is the pointed cone formed by the set of nonnegative elements in an ordered
+module. -/
+def positive : PointedCone 𝕜 E :=
+  (ConvexCone.positive 𝕜 E).toPointedCone <| ConvexCone.pointed_positive 𝕜 E
+
+@[simp]
+theorem mem_positive {x : E} : x ∈ positive 𝕜 E ↔ 0 ≤ x :=
+  Iff.rfl
+
+@[simp, norm_cast]
+theorem coe_positive : ↑(positive 𝕜 E) = ConvexCone.positive 𝕜 E :=
```

Comment:
I think this (and any other lemmas about `↑` that refer to `ConvexCone`) should be called
```suggestion
theorem toConvexCone_positive : ↑(positive 𝕜 E) = ConvexCone.positive 𝕜 E :=
```
so that `coe_` can be reserved for the statements about coercion to a set.